patches/tModLoader/Terraria/Program.cs.patch

```diff
@@ -100,16 +104,41 @@
  				args = Utils.ConvertMonoArgsToDotNet(args);
  
 +			LaunchParameters = Utils.ParseArguements(args);
++			
 +			SavePath =
 +				LaunchParameters.ContainsKey("-tmlsavedirectory") ? LaunchParameters["-tmlsavedirectory"] :
-+				LaunchParameters.ContainsKey("-savedirectory") ? Path.Combine(LaunchParameters["-savedirectory"], "ModLoader") :
-+				Platform.Get<IPathService>().GetStoragePath($"Terraria{Path.DirectorySeparatorChar}ModLoader{(!BuildInfo.IsRelease ? Path.DirectorySeparatorChar + "Beta" : "")}");
++				LaunchParameters.ContainsKey("-savedirectory") ? LaunchParameters["-savedirectory"] :
++				Platform.Get<IPathService>().GetStoragePath($"Terraria");
++
++			// Port old file format users
++			var oldBetas = Path.Combine(SavePath, "ModLoader/Beta");
++			
++			if (Directory.Exists(oldBetas)) {
++				var newPath = Path.Combine(SavePath, PreviewFolder);
++				Directory.Move(oldBetas, newPath);
++
++				string[] subDirsToMove = { "Mod Reader", "Mod Sources", "Mod Configs" };
++				foreach (var subDir in subDirsToMove) {
++					if (Directory.Exists(Path.Combine(newPath, subDir)))
++						Directory.Move(Path.Combine(newPath, subDir), Path.Combine(newPath, subDir.Replace(" ","")));
++				}
++
++				if (Directory.Exists(Path.Combine(newPath, "Workshop")))
++					Directory.Move(Path.Combine(newPath, "Workshop"), Path.Combine(SavePath, "Workshop"));
++
++				FileUtilities.CopyFolder(newPath, Path.Combine(SavePath, DevFolder));
++				FileUtilities.CopyFolder(newPath, Path.Combine(SavePath, ReleaseFolder));
++			}
++
++			var fileFolder =
++				BuildInfo.IsRelease ? ReleaseFolder :
++				BuildInfo.IsBeta ? PreviewFolder :
++				DevFolder;
++			SavePath = Path.Combine(SavePath, fileFolder);
```

Comment:
this should definitely be its own function

Reply:
You'll also need `try/catch` around all the file operations, in case of file in use errors. We should probably just display an error dialog if we can't move the dirs

Reply:
If we centralize the logs folder, this can also be done after logging is initialized.